app/packages/core/src/components/Actions/Similar.tsx

```diff
@@ -237,91 +291,201 @@ const SortBySimilarity = React.memo(
       current && setState(current);
     }, [current]);
 
+    useEffect(() => {
+      if (!hasNoSelectedSamples) {
+        setIsImageSearch(true);
+      }
+    }, [hasNoSelectedSamples]);
```

Comment:
This should be a selector. Right now this causes an unnecessary re-render

Reply:
will update this